File: backend/app/schemas/tax_summary.py
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, Field

from app.core.tax_rules_registry import MANDATORY_TAX_DISCLAIMER


class IncomeSummary(BaseModel):
    gross_salary: Decimal = Field(default=Decimal("0.00"))
    business_income: Decimal = Field(default=Decimal("0.00"))
    dividend_income: Decimal = Field(default=Decimal("0.00"))
    other_income: Decimal = Field(default=Decimal("0.00"))
    total_gross_income: Decimal = Field(default=Decimal("0.00"))
    total_tds_credits: Decimal = Field(default=Decimal("0.00"))


class ExemptionsSummary(BaseModel):
    standard_deduction: Decimal = Field(default=Decimal("0.00"))
    hra_exemption: Decimal = Field(default=Decimal("0.00"))
    professional_tax: Decimal = Field(default=Decimal("0.00"))
    children_education_allowance: Decimal = Field(default=Decimal("0.00"))
    employer_nps: Decimal = Field(default=Decimal("0.00"))
    total_exemptions: Decimal = Field(default=Decimal("0.00"))


class DeductionSummary(BaseModel):
    section_80c: Decimal = Field(default=Decimal("0.00"))
    section_80d: Decimal = Field(default=Decimal("0.00"))
    section_80ccd_1b: Decimal = Field(default=Decimal("0.00"))
    section_80e: Decimal = Field(default=Decimal("0.00"))
    section_80g: Decimal = Field(default=Decimal("0.00"))
    section_80tta_80ttb: Decimal = Field(default=Decimal("0.00"))
    other_deductions: Decimal = Field(default=Decimal("0.00"))
    total_chapter_via_deductions: Decimal = Field(default=Decimal("0.00"))


class CapitalGainsSummary(BaseModel):
    stcg_taxable: Decimal = Field(default=Decimal("0.00"))
    ltcg_taxable: Decimal = Field(default=Decimal("0.00"))
    stcg_tax: Decimal = Field(default=Decimal("0.00"))
    ltcg_tax: Decimal = Field(default=Decimal("0.00"))
    total_capital_gains_tax: Decimal = Field(default=Decimal("0.00"))


class RegimeCalculation(BaseModel):
    regime_type: str  # "OLD" or "NEW"
    gross_income: Decimal = Field(default=Decimal("0.00"))
    exemptions: Decimal = Field(default=Decimal("0.00"))
    chapter_via_deductions: Decimal = Field(default=Decimal("0.00"))
    taxable_income: Decimal = Field(default=Decimal("0.00"))
    tax_on_slabs: Decimal = Field(default=Decimal("0.00"))
    capital_gains_tax: Decimal = Field(default=Decimal("0.00"))
    section_87a_rebate: Decimal = Field(default=Decimal("0.00"))
    tax_after_rebate: Decimal = Field(default=Decimal("0.00"))
    cess: Decimal = Field(default=Decimal("0.00"))
    total_tax_liability: Decimal = Field(default=Decimal("0.00"))
    tds_credits: Decimal = Field(default=Decimal("0.00"))
    net_tax_payable: Decimal = Field(default=Decimal("0.00"))


class TaxSummaryResponse(BaseModel):
    financial_year: str
    user_id: int
    income_summary: IncomeSummary
    exemptions_summary_old: ExemptionsSummary
    exemptions_summary_new: ExemptionsSummary
    deduction_summary: DeductionSummary
    capital_gains_summary: CapitalGainsSummary
    old_regime: RegimeCalculation
    new_regime: RegimeCalculation
    recommended_regime: str  # "OLD" or "NEW"
    tax_savings: Decimal = Field(default=Decimal("0.00"))
    disclaimer: str = Field(default=MANDATORY_TAX_DISCLAIMER)

File: frontend/src/types/tax_summary.ts
export interface IncomeSummary {
  gross_salary: number;
  business_income: number;
  dividend_income: number;
  other_income: number;
  total_gross_income: number;
  total_tds_credits: number;
}

export interface ExemptionsSummary {
  standard_deduction: number;
  hra_exemption: number;
  professional_tax: number;
  children_education_allowance: number;
  employer_nps: number;
  total_exemptions: number;
}

export interface DeductionSummary {
  section_80c: number;
  section_80d: number;
  section_80ccd_1b: number;
  section_80e: number;
  section_80g: number;
  section_80tta_80ttb: number;
  other_deductions: number;
  total_chapter_via_deductions: number;
}

export interface CapitalGainsSummary {
  stcg_taxable: number;
  ltcg_taxable: number;
  stcg_tax: number;
  ltcg_tax: number;
  total_capital_gains_tax: number;
}

export interface RegimeCalculation {
  regime_type: 'OLD' | 'NEW';
  gross_income: number;
  exemptions: number;
  chapter_via_deductions: number;
  taxable_income: number;
  tax_on_slabs: number;
  capital_gains_tax: number;
  section_87a_rebate: number;
  tax_after_rebate: number;
  cess: number;
  total_tax_liability: number;
  tds_credits: number;
  net_tax_payable: number;
}

export interface TaxSummaryResponse {
  financial_year: string;
  user_id: number;
  income_summary: IncomeSummary;
  exemptions_summary_old: ExemptionsSummary;
  exemptions_summary_new: ExemptionsSummary;
  deduction_summary: DeductionSummary;
  capital_gains_summary: CapitalGainsSummary;
  old_regime: RegimeCalculation;
  new_regime: RegimeCalculation;
  recommended_regime: 'OLD' | 'NEW';
  tax_savings: number;
  disclaimer: string;
}
